fix(mangler, prettier, ast_tools): remove methods which are unstable in our MSRV

overlookmotel opened the pull request:
Prompted by #8913, I ran `cargo check` on Rust 1.81.0 and found some other places where we use language features which are not stable in our MSRV. This PR fixes them.

Note: The `#[expect]` attribute did not become stable until 1.81.0, and we use it liberally throughout the codebase. So Oxc still won't compile at present on our advertised MSRV (1.80.0).

overlookmotel commented:
* **#8929** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8929?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8929?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8929).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Feb 7, 6:51 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(mangler, prettier, ast_tools): remove methods which are unstable in our MSRV (#8929)

Prompted by #8913, I ran `cargo check` on Rust 1.81.0 and found some other places where we use language features which are not stable in our MSRV. This PR fixes them.

Note: The `#[expect]` attribute did not become stable until 1.81.0, and we use it liberally throughout the codebase. So Oxc still won't compile at present on our advertised MSRV (1.80.0).
